runtime/ast/type.go

```diff
@@ -570,13 +574,24 @@ func (t *ReferenceType) EndPosition(memoryGauge common.MemoryGauge) Position {
 	return t.Type.EndPosition(memoryGauge)
 }
 
-const referenceTypeAuthKeywordSpaceDoc = prettier.Text("auth ")
+const referenceTypeAuthKeywordDoc = prettier.Text("auth")
 const referenceTypeSymbolDoc = prettier.Text("&")
 
 func (t *ReferenceType) Doc() prettier.Doc {
 	var doc prettier.Concat
-	if t.Authorized {
-		doc = append(doc, referenceTypeAuthKeywordSpaceDoc)
+	if t.Authorization != nil {
+		doc = append(doc, referenceTypeAuthKeywordDoc)
+		if t.Authorization.EntitlementSet != nil && len(t.Authorization.EntitlementSet.Entitlements()) > 0 {
+			doc = append(doc, prettier.Text("("))
+			for i, entitlement := range t.Authorization.EntitlementSet.Entitlements() {
+				doc = append(doc, entitlement.Doc())
+				if i < len(t.Authorization.EntitlementSet.Entitlements())-1 {
+					doc = append(doc, prettier.Text(t.Authorization.EntitlementSet.Separator()), prettier.Space)
+				}
+			}
```

Comment:
Maybe refactor the repeated `t.Authorization.EntitlementSet.Entitlements()` expressions into a variable.